### PR TITLE
Add !whois.setdefault

### PR DESCRIPTION
**Pull request by BenSimner**
Adds new commands: whois.setdefault, whois.unset and whois.unsetdefault along with tests for each.

This is one of those instances where having tests that operated on some pre-defined MongoDB instance with some old data in it good, to ensure a change doesn't break already-existing data for smallish changes that don't require a full database migration (like this)

### CODE REVIEW

File: csbot/test/test_plugin_whois.py [hunk 1]
```diff
@@ -94,4 +129,38 @@ def test_client_reply_whois_multiple_users_channels(self):
     def test_client_reply_whois_self(self):
         yield from self._recv_privmsg('Nick!~user@host', '#First', '!whois.set test1')
         yield from self._recv_privmsg('Nick!~user@host', '#First', '!whois')
-        self.assert_sent('NOTICE {} :{}'.format('#First', 'Nick: test1'))
\ No newline at end of file
+        self.assert_sent('NOTICE {} :{}'.format('#First', 'Nick: test1'))
+
+    @failsafe
+    @run_client
+    def test_client_reply_whois_setdefault_unset(self):
```
Comment (alanbriolat) [inline csbot/test/test_plugin_whois.py]: Looks like this method name is duplicated by the method below it, and it doesn't describe this one properly? Also seems unlikely this test runs, because of the method getting overwritten.

File: csbot/plugins/whois.py [hunk 2]
```diff
@@ -44,20 +55,42 @@ def whois(self, e):
...
+            acc = {'account': user['account']}
         else:
-            return {'nick': nick,
-                    'channel': channel}
+            acc = {'nick': nick}
+
+        if channel is not None:
+            acc['channel'] = channel
+        else:
+            acc['default'] = 'true'
```
Comment (alanbriolat) [inline csbot/plugins/whois.py]: What would be so bad about letting `channel == None` mean that it's a non-channel-specific whois, i.e. the default, in the database too? Then you don't have to change so many things, it just kinda works automatically since you're already passing around `channel=None` to mean "default".

File: csbot/plugins/whois.py [hunk 3]
```diff
@@ -11,26 +11,37 @@ class Whois(Plugin):
...
         user = db.find_one(ident)
 
         if user is None:
-            return None
+            # try lookup the default one
+            ident = self.identify_user(nick)
+            user = db.find_one(ident)
+            if user is None:
+                return None
+            return user['data']
```
Comment (alanbriolat) [inline csbot/plugins/whois.py]: This bit is hard to follow now. Maybe refactor as a loop over the alternatives in preference order? E.g.
```python
for ident in (self.identify_user(nick, channel),  # Channel-specific
              self.identify_user(nick)):          # Default
    user = db.find_one(ident)
    if user:
        return user['data']
return None
```

File: csbot/test/test_plugin_whois.py (hunk 1 above)
Comment (BenSimner) [inline csbot/test/test_plugin_whois.py]: Overzealous copy&paste! Will update name to something more useful.

File: csbot/plugins/whois.py (hunk 3 above)
Comment (BenSimner) [inline csbot/plugins/whois.py]: Yeah I thought about implementing it another way but felt imperative code was clearer, but I do think your for loop is nicer. Maybe I'll look at places in the actual command defs where similar horrible logic is happening and do the same.

File: csbot/plugins/whois.py (hunk 2 above)
Comment (BenSimner) [inline csbot/plugins/whois.py]: I feel like you're misreading this code? It doesn't insert a channel for `channel = None` situations, I think maybe inserting None here would be sensible but I'm unsure what pymongo does with that, I assume casts it into the BSON null value?

**Comment by alanbriolat** [comment]
Looks good 👍
